### ISSUE TITLE
Alacritty drawing lag when switching to terminal or when starting in a tiling window manager

### ISSUE THREAD

**Issue by vlashada** [alacritty/alacritty#6327]
I am using Alacritty with DWM on NixOS. My problem is that there seems to be some lag when opening a new window or switching from a different workspace to one with Alacritty. I looks like the rendering might be delayed. Sometimes, if I already have an Alacritty instance open, for instance running neovim, and I open a new one, the new instance of Alacritty will have ghosting with the content from the other instance showing up in the other terminal too. If I switch to a workspace with Alacritty from one without, I do not get ghosting, but the terminal is unresponsive for a short while. I am able to type characters, and they show up after the delay, and they seem to always register, but if I quickly switch to an instance running neovim, and I press escape to exit insert mode, that will not always be registered.

I have tried other terminals and GUI programs, and the problem seems to only exist for Alacritty.

I'll be happy to provide more information if anything is unclear.

### System

OS: NixOS
Version: 0.10.1 ()
Linux/BSD: DWM (currently without a compositor, but I have also tried using picom, with same result)

### Logs

Font/Terminal size, Keyboard and bindings: [link to output from alacritty](https://gist.github.com/vlashada/a6a4466f23b6698e68b81cadd5372d5d)

**Comment by kchibisov** [comment]
Are you using nvidia drivers? Might be something messing up there, since if you're using instances alacritty can't know about other applications...

**Comment by vlashada** [comment]
No, I have a thinkpad laptop, and only integrated intel graphics.

**Comment by chrisduerr** [comment]
Do you have any other window managers available for testing? DWM isn't exactly known for its elaborate software support and quality.

**Comment by kchibisov** [comment]
Might be fixed with the latest master, since vsync is no longer used. But intel drivers are sort of buggy.

**Comment by jakubreron** [comment]
Same issue for me with Intel HD Graphics 3000 on ThinkPad T420, using Arch with DWM, however, I cannot reproduce the issue when I launch alacritty with vblank_mode=0 env variable (it forces vsync to be off)
